xbmc/windowing/gbm/DRMAtomic.cpp

```diff
@@ -22,11 +22,31 @@
 #include <unistd.h>
 
 using namespace KODI::WINDOWING::GBM;
-
+uint32_t CDRMAtomic::GetScalingFactor(uint32_t destWidth, uint32_t destHeight,uint32_t srcWidth, uint32_t srcHeight ){
+  uint32_t fW, fH;
+  fW = destWidth / srcWidth;
+  fH = destHeight / srcHeight;
+  if (fW != fH)
+     return (fW < fH) ? fW : fH;
+  return fW;
+}
+bool CDRMAtomic::SetScalingFilter(struct drm_object *object, const char *name) {
+  uint32_t scale_factor;
+  for (uint32_t i = 0; i < object->props->count_props; i++)
```

Comment:
Please use the `SupportsProperty` and `SupportsPropertyAndValue` helper functions here